laws/shared/src/main/scala/algebra/laws/RingLaws.scala

```diff
@@ -37,103 +37,144 @@ trait RingLaws[A] extends GroupLaws[A] {
 
   // multiplicative groups
 
-  def multiplicativeSemigroup(implicit A: MultiplicativeSemigroup[A]) = new MultiplicativeProperties(
-    base = _.semigroup(A.multiplicative),
-    parent = None,
-    Rules.serializable(A),
-    Rules.repeat1("pow")(A.pow),
-    Rules.repeat2("pow", "*")(A.pow)(A.times)
-  )
-
-  def multiplicativeMonoid(implicit A: MultiplicativeMonoid[A]) = new MultiplicativeProperties(
-    base = _.monoid(A.multiplicative),
-    parent = Some(multiplicativeSemigroup),
-    Rules.repeat0("pow", "one", A.one)(A.pow),
-    Rules.collect0("product", "one", A.one)(A.product)
-  )
-
-  def multiplicativeCommutativeMonoid(implicit A: MultiplicativeCommutativeMonoid[A]) = new MultiplicativeProperties(
-    base = _.commutativeMonoid(A.multiplicative),
-    parent = Some(multiplicativeMonoid)
-  )
-
-  def multiplicativeGroup(implicit A: MultiplicativeGroup[A]) = new MultiplicativeProperties(
-    base = _.group(A.multiplicative),
-    parent = Some(multiplicativeMonoid),
-    // pred is used to ensure y is not zero.
-    "consistent division" -> forAll { (x: A, y: A) =>
-      pred(y) ==> (A.div(x, y) ?== A.times(x, A.reciprocal(y)))
-    }
-  )
+  def multiplicativeSemigroup(implicit A: MultiplicativeSemigroup[A],
+    check: IsSerializable[Semigroup[A]],
+    multCheck: IsSerializable[MultiplicativeSemigroup[A]]) =
+    new MultiplicativeProperties(
+      base = _.semigroup(A.multiplicative, check),
+      parent = None,
+      Rules.serializable(A),
+      Rules.repeat1("pow")(A.pow),
+      Rules.repeat2("pow", "*")(A.pow)(A.times)
+    )
+
+  def multiplicativeMonoid(implicit A: MultiplicativeMonoid[A],
+    check: IsSerializable[Monoid[A]],
+    multCheck: IsSerializable[MultiplicativeMonoid[A]]) =
+    new MultiplicativeProperties(
+      base = _.monoid(A.multiplicative, check),
+      parent = Some(multiplicativeSemigroup),
+      Rules.repeat0("pow", "one", A.one)(A.pow),
+      Rules.collect0("product", "one", A.one)(A.product)
+    )
+
+  def multiplicativeCommutativeMonoid(implicit A: MultiplicativeCommutativeMonoid[A],
+    check: IsSerializable[CommutativeMonoid[A]],
+    multCheck: IsSerializable[MultiplicativeCommutativeMonoid[A]]) =
+    new MultiplicativeProperties(
+      base = _.commutativeMonoid(A.multiplicative, check),
```

Comment:
cases like this are weird too, since really what you want to do is derive from check. Like if `check == IsNot` then `multCheck` should be IsNot. Maybe just a method to change the type would be fine, and never really unsafe since this is just controlling test behavior.